src/Controller/HearingController.php

```diff
@@ -65,11 +66,17 @@ public function index(CaseEntity $case, HearingPostRepository $hearingPostReposi
             $this->entityManager->flush();
         }
 
-        $partyHasSomethingToAdd = $hasCounterparty
-            ? true === $hearing->getPartyHasNoMoreToAdd() && true === $hearing->getCounterpartHasNoMoreToAdd()
-            : true === $hearing->getPartyHasNoMoreToAdd();
+        $partyHasSomethingToAdd = false;
 
-        $form = $this->createForm(HearingFinishType::class, $hearing, ['case' => $case, 'hasCounterparty' => $hasCounterparty]);
+        if ($hasCounterparty && $hasParty) {
+            $partyHasSomethingToAdd = !($hearing->getCounterpartHasNoMoreToAdd() && $hearing->getPartyHasNoMoreToAdd());
```

Comment:
I think you mean `!… && !…` here (which is different from `!(… && …)`, cf. https://math.stackexchange.com/a/3109769).

Reply:
I think the confusion comes from my naming and the fact that `$partyHasSomethingToAdd` is the complementary to the information we have, i.e. `->getCounterpartHasNoMoreToAdd()`.

So what we need is `$partyHasSomethingToAdd` to become `true` if either of the `...HasNoMoreToAdd()` is `false`, i.e. `!… or !…`.